### PR TITLE
Refactor create function code action

### LINKED ISSUE #33138: `Create function` code action adds a new line unnecessarily to the end of the newly created function
**Description:**

1) Consider the scenario below, where there's a new line after the undefined function.
```
int i = getInt();

string s = "hello world";
```
`Create function` code action adds a new line at the end of the created function `getInt()`, which is not really necessary in this case.

Screenshot 2021-10-08 at 15 46 16

Can we add it only if there's no new line after the undefined function?

For example, in this case it would be necessary.
```
int i = getInt();
string s = "hello world";
```
2) There's an existing check in `CreateFunctionExecutor.java` that checks whether the file ends with a new line, but that doesn't seem to be needed now since it was only relevant when the newly created function was added to the end of the file.

```
// If the file ends with a new line
        boolean newLineAtEnd = rootLineRange.endLine().offset() == 0;
```

Suggestion to refactor the code here.

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject to  action to conditionally  add a new line at the end of the generated function.

Fixes #33138 

## Approach
Check if the immediate line after the enclosing node is occupied by a module member node.

## Samples

Peek 2022-03-22 12-50

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java [hunk 1]
```diff
@@ -173,11 +175,10 @@ public Object execute(ExecuteCommandContext context) throws LSCommandExecutorExc
...
         if (enclosingNode.isPresent()) {
-            LineRange endLineRange = enclosingNode.get().lineRange();
-            newLineAtEnd = false;
-            insertRange = new Range(new Position(endLineRange.endLine().line(), endLineRange.endLine().offset()),
-                    new Position(endLineRange.endLine().line(), endLineRange.endLine().offset()));
-
+            LineRange lineRange = enclosingNode.get().lineRange();
+            newLineAtEnd = addNewLineAtEnd(enclosingNode.get());
+            insertRange = new Range(new Position(lineRange.endLine().line(), lineRange.endLine().offset()),
+                    new Position(lineRange.endLine().line(), lineRange.endLine().offset()));
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java:181]: Shall we use `CommonUtil.toRange()` here?

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java:181]: We can't use that here, as we want the Range to be starting from the endLine and offset.

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java:181]: Then shall we add separate util to accept the line?

Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java]: Done

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**